Challenges/challenge_46_can_place_flowers/greghilston/src/main/java/Solution.java

```diff
@@ -0,0 +1,54 @@
+public class Solution  {
+    /**
+     * Determines if we can plant a flower at a given location in a specific flowerbed.
+     * @param flowerBed state of flowerbed
+     * @param i location to check
+     * @return whether we can plant a flower or not
+     */
+    private boolean canPlantHere(int[] flowerBed, int i) {
+        // left check
+        int leftNeighborIndex = i - 1;
+
+        if(leftNeighborIndex > 0) // ensures we don't go off the flower bed
+        {
+            if(flowerBed[leftNeighborIndex] == 1) {
+                return false; // Can't plant next to a flower
+            }
+        }
+
+        // right check
+        int rightNeighborIndex = i + 1;
+
+        if(rightNeighborIndex < flowerBed.length) // ensures we don't go off the flower bed
+        {
+            if(flowerBed[rightNeighborIndex] == 1) {
+                return false; // Can't plant next to a flower
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Determines if we can plant n new flowers in a specific flowerbed.
+     * @param flowerBed initial state of flowerbed
+     * @param n how many flowers we wish to plant
+     * @return whether we can plant n new flowers in flowerBed
+     */
+    public boolean solve(int[] flowerBed, int n) {
+        int planted = 0;
```

Comment:
[Nano-optimization] You can omit this additional local variable by just counting down `n` until it reaches `0` instead. In terms of readability, the additional variable is not a bad idea. But then again, the parameter is named `n`, so... :)

Reply:
lolololol. I understand what you're saying and I'll keep that in mind in the future but will leave this the way it is written for right now. Neat comment